integration-tests/tests/utils/config.py

```diff
@@ -0,0 +1,50 @@
+import shutil
+from dataclasses import dataclass
+from pathlib import Path
+
+
+@dataclass(frozen=True)
+class PackageConfig:
+    clp_bin_dir: Path
+    clp_package_dir: Path
+    clp_sbin_dir: Path
+    test_output_dir: Path
+    uncompressed_logs_dir: Path
+
+
+@dataclass(frozen=True)
+class DatasetLogs:
+    dataset_name: str
+    tarball_url: str
+    download_dir: Path
+    extraction_dir: Path
+    compression_dir: Path
+    decompression_dir: Path
+
+    @classmethod
+    def create(cls, package_config, dataset_name: str, **kwargs):
+        dataset_name = dataset_name.strip()
+        if 0 == len(dataset_name):
+            raise ValueError("`dataset_name` cannot be empty.")
+        return cls(
+            dataset_name=dataset_name,
+            tarball_url=kwargs.get("tarball_url", ""),
+            download_dir=kwargs.get(
```

Comment:
the second argument shouldn't be a filename if the variable name is download_"dir"?